docs/configuration.rst
Configuration
=======================
The global settings for ``pyAS2`` are kept in a single configuration dictionary named ``PYAS2`` in the ``settings.py`` module.
Below is a sample configuration:

.. code-block:: python

    PYAS2 = {
        'ENVIRONMENT' : 'production', 
        'PORT' : 8888, 
        'SSLCERTIFICATE' : '/path_to_cert/server_cert.pem', 
        'SSLPRIVATEKEY' : '/path_to_cert/server_privkey.pem',
        'DATADIR' : '/path_to_datadir/data', 
        'PYTHONPATH' : '/path_to_python/python', 
        'ENVIRONMENTTEXT' : 'BETA',  
        'ENVIRONMENTTEXTCOLOR' : 'Yellow', 
        'LOGLEVEL' : 'DEBUG', 
        'LOGCONSOLE' : True, 
        'LOGCONSOLELEVEL' : 'DEBUG', 
        'MAXRETRIES': 5,    
        'MDNURL' : 'https://192.168.1.115:8888/pyas2/as2receive', 
        'ASYNCMDNWAIT' : 30,
        'MAXARCHDAYS' : 30, 
    }

The available settings along with their usage is described below:

+------------------------+----------------------------+------------------------------------------------+
| Settings Name          | Default Value              | Usage                                          |
+========================+============================+================================================+
| ENVIRONMENT            | production                 | The as2 server in development or production    |
+------------------------+----------------------------+------------------------------------------------+
| PORT                   | 8080                       | HTTP Port as2 server listens on                |
+------------------------+----------------------------+------------------------------------------------+
| SSLCERTIFICATE         | ``None``                   | Path to the SSL Public Key                     |
+------------------------+----------------------------+------------------------------------------------+
| SSLPRIVATEKEY          | ``None``                   | Path to the SSL Private Key                    |
+------------------------+----------------------------+------------------------------------------------+
| DATADIR                | ``Django Project Path``    | Full path to the base directory for storing    | 
|                        |                            | messages, MDNs, certificates and logs          |
+------------------------+----------------------------+------------------------------------------------+
| PYTHONPATH             | ``System Python Path``     | Path to the python executable, required with   |
|                        |                            | virtual environments                           |
+------------------------+----------------------------+------------------------------------------------+
| ENVIRONMENTTEXT        | ``None``                   | Text displayed on right of the logo. Useful    |
|                        |                            | to indicate different environments.            |
+------------------------+----------------------------+------------------------------------------------+
| ENVIRONMENTTEXTCOLOR   | Black                      | Color of the displayed PYTHONPATH. Use HTML    | 
|                        |                            | valid "color name" or #RGB values.             |
+------------------------+----------------------------+------------------------------------------------+
| LOGLEVEL               | INFO                       | Level for logging to log file. Values:         |
|                        |                            | DEBUG,INFO,STARTINFO,WARNING,ERROR or CRITICAL.| 
+------------------------+----------------------------+------------------------------------------------+
| LOGCONSOLE             | True                       | Console logging on (True) or off (False).      |
+------------------------+----------------------------+------------------------------------------------+
| LOGCONSOLELEVEL        | STARTINFO                  | level for logging to console/screen. Values:   | 
|                        |                            | DEBUG,INFO,STARTINFO,WARNING,ERROR or CRITICAL.| 
+------------------------+----------------------------+------------------------------------------------+
| MAXRETRIES             | 10                         | Maximum number of retries for failed outgoing  |
|                        |                            | messages                                       |
+------------------------+----------------------------+------------------------------------------------+
| MDNURL                 | ``None``                   | Return URL for receiving asynchronous MDNs from|
|                        |                            | partners.                                      |
+------------------------+----------------------------+------------------------------------------------+
| ASYNCMDNWAIT           | 30                         | Number of minutes to wait for asynchronous MDNs| 
|                        |                            | after which message will be marked as failed.  |
+------------------------+----------------------------+------------------------------------------------+
| MAXARCHDAYS            | 30                         | Number of days files and messages are kept in  |
|                        |                            | storage.                                       |
+------------------------+----------------------------+------------------------------------------------+


CHANGELOG.rst
Release History
===============

0.2 - `master`_
~~~~~~~~~~~~~~~

.. note:: This version is not yet released and is under active development.

0.1.2 - 2014-09-07
~~~~~~~~~~~~~~~~~~

* Created readthedocs documentation.
* Fixed bug where inbox and outbox folders were not created on saving partners and orgs.
* Fixed bug where MDN search was failing due to orphaned MDNs.

0.1.1 - 2015-09-04
~~~~~~~~~~~~~~~~~~

* Increased the max length of MODE_CHOICES model field.
* Detect Signature Algorithm from the MIME message for outbound messages.

0.1 - 2015-04-29
~~~~~~~~~~~~~~~~

* Initial release.

.. _`master`: https://github.com/abhishek-ram/pyas2 


djproject/settings.py
"""
Django settings for djproject project.

For more information on this file, see
https://docs.djangoproject.com/en/1.7/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/1.7/ref/settings/
"""

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
import os
BASE_DIR = os.path.dirname(os.path.dirname(__file__))


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/1.7/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = '7slz5$2)6k0%)(mip*mw9aa836al=+6!5dz23lan)p0@=*597a'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

TEMPLATE_DEBUG = True

ALLOWED_HOSTS = []

TEMPLATE_DIRS = (os.path.join(BASE_DIR, 'pyas2/templates'),)

# Application definition

INSTALLED_APPS = (
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'pyas2',
)

MIDDLEWARE_CLASSES = (
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
)

ROOT_URLCONF = 'djproject.urls'

WSGI_APPLICATION = 'djproject.wsgi.application'

# Database
# https://docs.djangoproject.com/en/1.7/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

# Internationalization
# https://docs.djangoproject.com/en/1.7/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/1.7/howto/static-files/

STATIC_URL = '/static/'

MANAGERS = (    #pyas2 will send error reports to the MANAGERS
        ('abhishek ram', 'abhishek8816@gmail.com'),
    )
EMAIL_HOST = 'localhost'             #Default: 'localhost'
EMAIL_PORT = '25'             #Default: 25
EMAIL_USE_TLS = False       #Default: False
EMAIL_HOST_USER = ''        #Default: ''. Username to use for the SMTP server defined in EMAIL_HOST. If empty, Django won't attempt authentication.
EMAIL_HOST_PASSWORD = ''    #Default: ''. PASSWORD to use for the SMTP server defined in EMAIL_HOST. If empty, Django won't attempt authentication.

## PYAS2 app settings
PYAS2 = {
    ## as2 server in development or production. default is production
    'ENVIRONMENT' : 'production',
    ## port the as2 server listens. default is 8080
    'PORT' : 8888,
    ## Specift in order to enable ssl/https for the as2 server:
    'SSLCERTIFICATE' : '/Users/abhishekram/Documents/work/Research/AS2App/server_cert.pem',
    'SSLPRIVATEKEY' : '/Users/abhishekram/Documents/work/Research/AS2App/server_privkey.pem',
    ## Full path to the base directory for storing files, logs ...
    'ROOTDIR' : os.path.dirname(os.path.dirname(__file__)),
    ## Path to the python executable, neccessary with virtual environments
    'PYTHONPATH' : '/opt/pyenv/python26/dj_latest/bin/python',
    ## environment_text: text displayed on right of the logo. Useful to indicate different environments.
    'ENVIRONMENTTEXT' : 'BETA',   
    ## environment_text_color: Use HTML valid "color name" or #RGB values. Default: Black (#000000)
    'ENVIRONMENTTEXTCOLOR' : 'Yellow',
    ## level for logging to log file. Values: DEBUG,INFO,STARTINFO,WARNING,ERROR or CRITICAL. Default: INFO
    'LOGLEVEL' : 'DEBUG',
    ## console logging on (True) or off (False); default is True.
    'LOGCONSOLE' : True,
    ## level for logging to console/screen. Values: DEBUG,INFO,STARTINFO,WARNING,ERROR or CRITICAL. Default: STARTINFO  
    'LOGCONSOLELEVEL' : 'DEBUG',
    ## Maximum number of retries for failed outgoing messages, defaule is 10
    'MAXRETRIES': 5,	
    ## Return url for receiving async MDNs from partners
    'MDNURL' : 'http://104.155.212.52:8001/pyas2/as2receive',
    ## Maximum wait time in minutes for asyn MDNs from partner, post which message will be marked as failed
    'ASYNCMDNWAIT' : 30,
    ## number of days files and messages are kept in storage; default is 30
    'MAXARCHDAYS' : 30,
}

